core/src/main/java/org/elasticsearch/script/Script.java

```diff
@@ -89,6 +93,88 @@
     public static final ParseField PARAMS_PARSE_FIELD = new ParseField("params");
 
     /**
+     * Constructs and validates a {@link Script} being parsed as XContent.
+     */
+    @SuppressWarnings("unchecked")
```

Comment:
Switched this over.  This is better because it doesn't require ConstructedObjectParser to take in dynamic default values.